gremlin-groovy/src/main/java/org/apache/tinkerpop/gremlin/groovy/engine/GremlinExecutor.java

```diff
@@ -269,7 +273,18 @@ public CompletableFuture<Object> eval(final String script, final String language
 
                 logger.debug("Evaluating script - {} - in thread [{}]", script, Thread.currentThread().getName());
 
-                final Object o = gremlinScriptEngineManager.getEngineByName(lang).eval(script, bindings);
+                Object o = null;
+                if (script instanceof String)
+                    o = gremlinScriptEngineManager.getEngineByName(lang).eval((String) script, bindings);
+                else if (script instanceof Bytecode) {
```

Comment:
Nit: Rename `script` to something more generic such as `message` since it could either be a script or bytecode